core/citrus-api/src/main/java/com/consol/citrus/variable/VariableExpressionIterator.java

```diff
@@ -0,0 +1,151 @@
+     */
+    private VariableSegment nextSegment;
+
+    public VariableExpressionIterator(String variableExpression, TestContext testContext, List<SegmentVariableExtractor> segmentValueExtractors) {
+        this.testContext = testContext;
+        this.segmentValueExtractors = segmentValueExtractors;
+
+        matcher = new VariableExpressionSegmentMatcher(variableExpression);
+
+        if (matcher.nextMatch()) {
+           nextSegment = createSegmentValue(testContext.getVariables());
+        } else {
+            throw new CitrusRuntimeException(
+                    String.format("Cannot match a segment on variableExpression: %s", variableExpression));
+        }
+    }
+
+    /**
+     * Returns true if the iterator has a next
+     * @return
+     */
+    @Override
+    public boolean hasNext() {
+        return nextSegment != null;
+    }
+
+    /**
+     * Returns the next value and looks ahead for yet another next value.
+     * @return
+     */
+    @Override
+    public VariableSegment next() {
+        VariableSegment currentSegment = nextSegment;
+
+        // Look ahead next segment
+        if (matcher.nextMatch()) {
+            nextSegment = createSegmentValue(currentSegment.getSegmentValue());
+        } else {
+            nextSegment = null;
+        }
+
+        return currentSegment;
+    }
+
+    /**
+     * Create the segment value from the current match
+     * @param parentValue
+     * @return
+     */
+    private VariableSegment createSegmentValue(Object parentValue) {
+
+        Object segmentValue = segmentValueExtractors.stream().filter(extractor->extractor.canExtract(testContext, parentValue, matcher)).findFirst().map(extractor->extractor.extractValue(testContext, parentValue, matcher)).orElse(null);
+        return new VariableSegment(matcher.getSegmentExpression(), matcher.getSegmentIndex(), segmentValue);
+    }
+
+    public static Object getLastExpressionValue(String variableExpression, TestContext testContext, List<SegmentVariableExtractor> extractors) {
+        VariableSegment segment = null;
+        VariableExpressionIterator iterator = new VariableExpressionIterator(variableExpression, testContext, extractors);
+        while (iterator.hasNext()) {
+            segment = iterator.next();
+        }
+
+        return segment != null ? segment.getSegmentValue() : null;
+    }
+
+    public static class VariableSegment {
+
+        /**
+         * The name of the variableExpression segment
+         */
+        private final String name;
+
+        /**
+         * An optional index if the VariableSegment represents an array. A value of -1 indicates "no index".
+         */
+        private final int index;
+
+        /**
+         * The evaluated value for this segment
+         */
+        private final Object segmentValue;
+
+        public VariableSegment(String name, int index, Object segmentValue) {
+            super();
```

Comment:
that's a useless `super` invocation. the file hasn't been formatted too 😉

Reply:
Removed that one. 

I also reformatted the code.

I now formatted the code using the standard settings of IntelliJ. Hope that meets your